scripts/dts/gen_defines.py

```diff
@@ -453,20 +453,20 @@ def map_arm_gic_irq_type(irq, irq_num):
             return irq_num + 16
         err(f"Invalid interrupt type specified for {irq!r}")
 
-    def encode_zephyr_multi_level_irq(irq, irq_num):
-        # See doc/kernel/services/interrupts.rst for details
-        # on how this encoding works
-
-        irq_ctrl = irq.controller
-        # Look for interrupt controller parent until we have none
-        while irq_ctrl.interrupts:
-            irq_num = (irq_num + 1) << 8
-            if "irq" not in irq_ctrl.interrupts[0].data:
-                err(f"Expected binding for {irq_ctrl!r} to have 'irq' in "
-                    "interrupt-cells")
-            irq_num |= irq_ctrl.interrupts[0].data["irq"]
-            irq_ctrl = irq_ctrl.interrupts[0].controller
-        return irq_num
+    # def encode_zephyr_multi_level_irq(irq, irq_num):
```

Comment:
I would remove the commented-out lines. Architecturally speaking, I think this proposal is sound - it's better to do this in C via DT macros than to have part of it live in python.